core/src/test/scala/unit/kafka/admin/DeleteOffsetsConsumerGroupCommandTest.scala

```diff
@@ -0,0 +1,185 @@
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.serialization.ByteArrayDeserializer
+import org.apache.kafka.common.serialization.ByteArraySerializer
+import org.apache.kafka.common.utils.Utils
+import org.junit.Test
+import org.junit.Assert._
+
+class DeleteOffsetsConsumerGroupCommandTest extends ConsumerGroupCommandTest {
+
+  def getArgs(group: String, topic: String): Array[String] = {
+    Array(
+      "--bootstrap-server", brokerList,
+      "--delete-offsets",
+      "--group", group,
+      "--topic", topic
+    )
+  }
+
+  @Test
+  def testDeleteOffsetsNonExistingGroup(): Unit = {
+    val group = "missing.group"
+    val topic = "foo:1"
+
+    val service = getConsumerGroupService(getArgs(group, topic))
+
+    try {
+      service.deleteOffsets(group, List(topic))
+      fail("GroupIdNotFoundException should have been raised")
+    } catch {
+      case e: ExecutionException =>
+        if (e.getCause != Errors.GROUP_ID_NOT_FOUND.exception())
+          throw e
+    }
+  }
+
+  @Test
+  def testDeleteOffsetsWithTopicPartition(): Unit = {
+    val producer = createProducer()
+    try {
+      producer.send(new ProducerRecord(topic, 0, null, null)).get()
+    } finally {
+      Utils.closeQuietly(producer, "producer")
+    }
+
+    val consumer = createConsumer()
+    try {
+      TestUtils.subscribeAndWaitForRecords(this.topic, consumer)
+      consumer.commitSync()
+
+      val topic = this.topic + ":0"
+
+      val service = getConsumerGroupService(getArgs(group, topic))
+
+      val partitions = service.deleteOffsets(group, List(topic))
+      // Unknown because the consumer has not committed any offsets yet.
+      assertEquals(Errors.GROUP_SUBSCRIBED_TO_TOPIC.exception, partitions(new TopicPartition(this.topic, 0)).getCause)
+    } finally {
+      Utils.closeQuietly(consumer, "consumer")
+    }
+  }
+
+  @Test
+  def testDeleteOffsetsWithTopic(): Unit = {
+    val producer = createProducer()
+    try {
+      producer.send(new ProducerRecord(topic, 0, null, null)).get()
+    } finally {
+      Utils.closeQuietly(producer, "producer")
+    }
+
+    val consumer = createConsumer()
+    try {
+      TestUtils.subscribeAndWaitForRecords(topic, consumer)
+      consumer.commitSync()
+
+      val service = getConsumerGroupService(getArgs(group, topic))
+
+      val partitions = service.deleteOffsets(group, List(topic))
+      assertEquals(Errors.GROUP_SUBSCRIBED_TO_TOPIC.exception, partitions(new TopicPartition(topic, 0)).getCause)
+    } finally {
+      Utils.closeQuietly(consumer, "consumer")
+    }
+  }
+
+  @Test
+  def testDeleteOffsetsWithTopicEmpty(): Unit = {
```

Comment:
What does empty topic mean here?

Reply:
Hum... I don't remember what I meant which indicated that the name is poor. I guess a word is missing here because it tests the case where a topic is provided with no partitions. I'll rename it.